### PR TITLE
Add support for loading Private Spaces using HF Token

### PR DESCRIPTION
**Pull request by abidlabs**
* Adds support for loading private Spaces, and fixes: #2557 
* Updates the Space URLs to use the new format: `{space_name}.hf.space`
* Adds some typing and minor refactoring to the `external.py` file. More refactoring to come in a separate PR

### CODE REVIEW

File: gradio/external.py [hunk 1]
```diff
@@ -457,8 +472,10 @@ def streamline_config(config: dict) -> dict:
         return config
 
     config = streamline_config(config)
-    api_url = "https://hf.space/embed/{}/api/predict/".format(model_name)
+    api_url = "{}/api/predict/".format(iframe_url)
     headers = {"Content-Type": "application/json"}
+    if api_key is not None:
+        headers["Authorization"] = f"Bearer {api_key}"
     ws_url = "wss://spaces.huggingface.tech/{}/queue/join".format(model_name)
```
Comment (freddyaboulton) [inline gradio/external.py:479]: This should be `iframe_url/queue/join` right?

File: gradio/blocks.py [hunk 2]
```diff
@@ -1054,10 +1054,10 @@ def load(
 
         Instance method: adds event that runs as soon as the demo loads in the browser. Example usage below.
         Parameters:
-            name: Class Method - the name of the model (e.g. "gpt2"), can include the `src` as prefix (e.g. "models/gpt2")
-            src: Class Method - the source of the model: `models` or `spaces` (or empty if source is provided as a prefix in `name`)
-            api_key: Class Method - optional api key for use with Hugging Face Hub
-            alias: Class Method - optional string used as the name of the loaded model instead of the default name
+            name: Class Method - the name of the model (e.g. "gpt2" or "facebook/bart-base") or space (e.g. "flax-community/spanish-gpt2"), can include the `src` as prefix (e.g. "models/facebook/bart-base")
+            src: Class Method - the source of the model: `models` or `spaces` (or leave empty if source is provided as a prefix in `name`)
+            api_key: Class Method - optional access token for loading private Hugging Face Hub models or spaces. Find your token here: https://huggingface.co/settings/tokens
```
Comment (freddyaboulton) [inline gradio/blocks.py:1059]: nit: Mention only read access required on the token?

**Review by freddyaboulton — APPROVED** [review]
Thanks for the fix @abidlabs !

File: gradio/external.py (hunk 1 above)
Comment (abidlabs) [reply gradio/external.py:479]: Yes I thought I fixed this, thank you!

File: gradio/blocks.py (hunk 2 above)
Comment (abidlabs) [reply gradio/blocks.py:1059]: Sure sounds good

**Comment by abidlabs** [comment]
Cool thanks for the review @freddyaboulton!

**Pull request merged**